programs/drift/src/state/lp_pool.rs

```diff
@@ -676,20 +681,33 @@ impl LPPool {
                     .cast::<i64>()?;
                 crypto_delta = crypto_delta.safe_add(constituent_target_notional.cast()?)?;
             }
-            aum = aum.safe_add(constituent_aum)?;
+            if constituent_aum < 0 {
+                aum = aum.saturating_sub(constituent_aum.abs().cast::<u128>()?);
```

Comment:
this won't work since its sensitive order of updates, positive ones need to be first (will be wrong if a large enough negative before a positive one)